ntbs-service/Pages/ServiceDirectory/SearchResults.cshtml

```diff
@@ -11,82 +11,143 @@
     <back-link-retaining-history inline-template>
         <nhs-back-link v-on:click="navigateBack" href="/ServiceDirectory">Back</nhs-back-link>
     </back-link-retaining-history>
-    
+
     <partial name="Partials/SearchFormPartial" model="Model"/>
 
     <div class="case-manager-results-summary">
         @{
-            var resultCount = Model.UserSearchResults?.NumberOfResults ?? 0;
+            var resultCount = Model.DirectorySearchResults?.NumberOfResults ?? 0;
             var resultsText = resultCount == 1 ? "result" : "results";
             var resultsFoundString = $"{resultCount.ToString()} {resultsText} found";
         }
         <p> @resultsFoundString </p>
     </div>
-    
-    @if (Model.UserSearchResults != null)
+
+    @if (Model.DirectorySearchResults is not null)
     {
         <ol class="search-results-list" id="search-results">
-            @foreach (var user in @Model.UserSearchResults)
+
+            @if (Model.RegionSearchResults is not null)
```

Comment:
Rather than having 4 separate models to iterate over, I think the structure here should be:
```
@foreach (var item in Model.DirectorySearchResults)
{
    @if (item.IsRegion())
    {
        // Display region
    }
    @if (item.IsService())
    // etc
}
```